components/AnchoredHeading.tsx

```diff
@@ -7,11 +7,42 @@ type AnchorHeadingProps = PropsWithChildren<{
   id?: string;
 }>;
 
+/**
+ * This module will automatically replace the header style words
+ * (such as `#`,`##`...ect) to an anchor name.
+ *
+ * This processor will make all the English characters (26 letters)
+ * into their lower case, for others in the middle of words,
+ * they will become `-` in default. For more than one `-` together,
+ * there's only one left.
+ *
+ * If your title has some non-English characters, please use
+ * `<!---->` to quote your English anchor name inside, with your
+ * own title beside it. Otherwises it will return you what it is.
+ */
+
+// React will ignore "<!" and ">" in the string,
+// so we can just use '-- --' to quote the anchor name inside it.
+const COMMENT_FOR_HEADANCHOR = /--\x20?([\w\x20-]+)\x20?--/;
+
 const AnchoredHeading = ({ children, level, id }: AnchorHeadingProps) => {
   const HeadingLevelTag = `h${level}` as any;
 
-  // This regex replacement is used for <!-- --> comments within the heading
-  const sanitizedId = id?.replace(/---(.+)---(.*)/, (_, f) => f);
+  let sanitizedId =
+    id ?? children?.toLocaleString().toLocaleLowerCase().replace(/\x20/g, '-');
+
+  if (sanitizedId) {
+    const foundAnchorAndTitle = COMMENT_FOR_HEADANCHOR.exec(sanitizedId);
```

Comment:
Can we add a white space here?

Reply:
Do you mean the white space shouldn't be converted to "-"?

I intend to do that, however I find it will be converted in default....So this is only a sync conversion :)

Reply:
Oh, I meant an empty line :) to separate this line to the next

Reply:
> Oh, I meant an empty line :) to separate this line to the next

Do you mean writing like this following:

```js
  let sanitizedId =
    id ?? 
    children?.toLocaleString().toLocaleLowerCase().replace(/\x20/g, '-');
```

When running "lint:js -- --fix", it automatically fix it like it....

If I get you wrong, please point it out :)

Reply:
No no, I literally meant adding a blank line between 35 and 36 😅

Reply:
Got it and fixed it, thanks for your patience XD